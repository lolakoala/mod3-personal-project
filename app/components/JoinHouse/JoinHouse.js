import React from 'react';
import PropTypes from 'prop-types';

const JoinHouse = ({ handleChange, getHouse, createHouse }) => {
  return (
    <div className='join-house'>
      <input type='text'
        placeholder="House Code"
        onChange={ (event) => handleChange(event, 'houseCode' ) }/>
      <button
        onClick={getHouse}>
          Join a House
      </button>
      <input type="text"
        placeholder="House Name"
        onChange={ (event) => handleChange(event, 'houseName' ) }/>
      <input type="text"
        placeholder="House Code"
        onChange={ (event) => handleChange(event, 'houseCode' ) }/>
      <button
        onClick={createHouse}>
          Add My House
      </button>
    </div>
  );
};

export default JoinHouse;

JoinHouse.propTypes = {
  handleChange: PropTypes.func,
  getHouse: PropTypes.func,
  createHouse: PropTypes.func
};
