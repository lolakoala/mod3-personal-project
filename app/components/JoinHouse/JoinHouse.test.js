import React from "react";
import JoinHouse from './JoinHouse.js';
import { shallow } from "enzyme";


describe('JoinHouse component', () => {
  const mockFn = jest.fn();
  const wrapper = shallow(<JoinHouse handleChange={mockFn}
    getHouse={mockFn}
    createHouse={mockFn}/>);

  it('should exist', () => {
    expect(wrapper).toBeDefined();
  });

  it('should have three inputs', () => {
    const inputs = wrapper.find('input');

    expect(inputs.length).toEqual(3);
  });

  it('should have 2 buttons', () => {
    const buttons = wrapper.find('button');

    expect(buttons.length).toEqual(2);
  });

  it('should fire actions', () => {
    const input = wrapper.find('input').first();
    const button1 = wrapper.find('button').first();
    const button2 = wrapper.find('button').last();

    input.simulate('change', { target: { value: 'woof9' } });
    button1.simulate('click');
    button2.simulate('click');
    expect(mockFn).toHaveBeenCalledTimes(3);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
