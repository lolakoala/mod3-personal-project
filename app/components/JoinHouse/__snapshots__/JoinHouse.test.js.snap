// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JoinHouse component should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <JoinHouse
    createHouse={[Function]}
    getHouse={[Function]}
    handleChange={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          onChange={[Function]}
          placeholder="House Code"
          type="text"
        />,
        <button
          disabled={true}
          onClick={[Function]}
        >
          Join a House
        </button>,
        <input
          onChange={[Function]}
          placeholder="House Name"
          type="text"
        />,
        <input
          onChange={[Function]}
          placeholder="House Code"
          type="text"
        />,
        <button
          disabled={true}
          onClick={[Function]}
        >
          Add My House
        </button>,
      ],
      "className": "join-house",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "onChange": [Function],
          "placeholder": "House Code",
          "type": "text",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Join a House",
          "disabled": true,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Join a House",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "onChange": [Function],
          "placeholder": "House Name",
          "type": "text",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "onChange": [Function],
          "placeholder": "House Code",
          "type": "text",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Add My House",
          "disabled": true,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Add My House",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            onChange={[Function]}
            placeholder="House Code"
            type="text"
          />,
          <button
            disabled={true}
            onClick={[Function]}
          >
            Join a House
          </button>,
          <input
            onChange={[Function]}
            placeholder="House Name"
            type="text"
          />,
          <input
            onChange={[Function]}
            placeholder="House Code"
            type="text"
          />,
          <button
            disabled={true}
            onClick={[Function]}
          >
            Add My House
          </button>,
        ],
        "className": "join-house",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "onChange": [Function],
            "placeholder": "House Code",
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Join a House",
            "disabled": true,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Join a House",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "onChange": [Function],
            "placeholder": "House Name",
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "onChange": [Function],
            "placeholder": "House Code",
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Add My House",
            "disabled": true,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Add My House",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
